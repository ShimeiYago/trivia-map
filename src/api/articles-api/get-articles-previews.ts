import { BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockGetArticlesPreviewsResponse } from '../mock/articles-response';
import { PaginationResponse } from 'api/types/pagination-response';
import { Park } from './../../types/park';
import { getUrlParameters } from 'utils/get-url-parameters';

export async function getArticlesPreviews(
  param: GetArticlesPreviewsParam,
): Promise<GetArticlesPreviewsResponse> {
  const axiosInstance = getAxiosInstance({}, mockGetArticlesPreviewsResponse);

  const urlParams = getUrlParameters(param);
  const url = `${BASE_URL}/articles/public/previews${urlParams}`;

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<GetArticlesPreviewsResponse> =
      await axiosInstance.get(url);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type GetArticlesPreviewsResponseEachItem = {
  postId: number;
  title: string;
  image: string | null;
  category: number;
};

export type GetArticlesPreviewsResponse =
  PaginationResponse<GetArticlesPreviewsResponseEachItem>;

export type GetArticlesPreviewsParam = {
  page?: number;
  marker?: number;
  user?: number;
  category?: number;
  park?: Park;
  keywords?: string[];
};
