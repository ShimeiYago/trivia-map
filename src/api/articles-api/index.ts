import { BASE_URL } from '../constants';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import {
  mockGetArticleResponse,
  mockPostArticleResponse,
} from '../mock/articles-response';
import { Position } from 'types/position';

export async function getRemoteArticle(
  postId: string,
): Promise<GetArticleResponse> {
  const axiosInstance = getAxiosInstance({}, mockGetArticleResponse);

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<GetArticleResponse> = await axiosInstance.get(
      `${BASE_URL}/articles/${postId}`,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export async function postRemoteArticle(
  title: string,
  content: string,
  position: Position,
): Promise<PostArticleResponse> {
  const axiosInstance = getAxiosInstance({}, mockPostArticleResponse);

  const requestData: PostArticleRequest = {
    title: title,
    content: content,
    position: position,
  };

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<PostArticleResponse> = await axiosInstance.post(
      `${BASE_URL}/articles`,
      requestData,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError<ValidationError>(axiosError);
  }
}

export async function putRemoteArticle(
  postId: string,
  title: string,
  content: string,
  position: Position,
): Promise<PostArticleResponse> {
  const axiosInstance = getAxiosInstance({}, mockPostArticleResponse);

  const requestData: PostArticleRequest = {
    title: title,
    content: content,
    position: position,
  };

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<PostArticleResponse> = await axiosInstance.put(
      `${BASE_URL}/articles/${postId}`,
      requestData,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError<ValidationError>(axiosError);
  }
}

export type GetArticleResponse = {
  title: string;
  content: string;
};

export type PostArticleRequest = {
  title: string;
  content: string;
  position: Position;
};

export type PostArticleResponse = {
  postId: string;
};

type ValidationError = {
  headerErrors: string[];
  fieldErrors?: {
    title?: string[];
    content?: string[];
    position?: string[];
  };
};
