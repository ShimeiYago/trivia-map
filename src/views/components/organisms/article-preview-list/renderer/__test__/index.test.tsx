import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, Props, State } from '..';
import * as GetArticlesPreviewsApiModule from 'api/articles-api/get-articles-previews';
import { mockGetArticlesPreviewsResponse } from 'api/mock/articles-response';

let wrapper: ShallowWrapper<Props, State, Renderer>;
let getArticlesPreviewsSpy: jest.SpyInstance;

const articlesPreviewsZero = {
  ...mockGetArticlesPreviewsResponse,
  results: [],
};

const articlesPreviewsOnePage = {
  ...mockGetArticlesPreviewsResponse,
  totalRecords: 2,
  totalPages: 1,
  currentPage: 1,
  startIndex: 1,
  endIndex: 2,
};

const basicProps: Props = {
  searchConditions: {
    marker: 1,
  },
  variant: 'large',
  throwError: jest.fn(),
  refreshUser: jest.fn(),
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    wrapper = shallow(<Renderer {...basicProps} />);
    getArticlesPreviewsSpy = jest.spyOn(GetArticlesPreviewsApiModule, 'getArticlesPreviews');
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('with articlesPreviews', () => {
    wrapper.setState({
      loadingState: 'success',
      articlesPreviews: mockGetArticlesPreviewsResponse,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('articlesPreviews is zero', () => {
    wrapper.setState({
      loadingState: 'success',
      articlesPreviews: articlesPreviewsZero,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('variant popup', () => {
    wrapper.setProps({ variant: 'popup' });
    wrapper.setState({
      loadingState: 'success',
      articlesPreviews: mockGetArticlesPreviewsResponse,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('variant sidebar', () => {
    wrapper.setProps({ variant: 'sidebar' });
    wrapper.setState({
      loadingState: 'success',
      articlesPreviews: mockGetArticlesPreviewsResponse,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('only one page with large', () => {
    wrapper.setProps({ variant: 'large' });
    wrapper.setState({
      loadingState: 'success',
      articlesPreviews: articlesPreviewsOnePage,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('only one page with popup', () => {
    wrapper.setProps({ variant: 'popup' });
    wrapper.setState({
      loadingState: 'success',
      articlesPreviews: articlesPreviewsOnePage,
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('fetchArticlesPreviews', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    getArticlesPreviewsSpy = jest.spyOn(GetArticlesPreviewsApiModule, 'getArticlesPreviews');
  });

  it('should set success states if api calling succeed', async () => {
    getArticlesPreviewsSpy.mockResolvedValue(mockGetArticlesPreviewsResponse);

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();
    await instance['fetchArticlesPreviews']();

    expect(instance.state.loadingState).toBe('success');
  });

  it('should call throwError if api calling fail', async () => {
    getArticlesPreviewsSpy.mockRejectedValue(new Error());

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();
    await instance['fetchArticlesPreviews']();

    expect(instance.props.throwError).toBeCalled();
  });
});

describe('handleChangePagination', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    getArticlesPreviewsSpy = jest.spyOn(GetArticlesPreviewsApiModule, 'getArticlesPreviews');
  });

  it('should set success states if api calling succeed', () => {
    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();
    instance['handleChangePagination']({} as React.ChangeEvent<unknown>, 1);

    expect(getArticlesPreviewsSpy).toBeCalled();
  });

  it('should scroll', () => {
    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    instance.topRef = {
      current: {
        scrollIntoView: jest.fn(),
      },
    } as unknown as React.RefObject<HTMLDivElement>;

    instance['handleChangePagination']({} as React.ChangeEvent<unknown>, 1);

    expect(instance.topRef.current?.scrollIntoView).toBeCalled();
  });
});

describe('componentDidUpdate', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    getArticlesPreviewsSpy = jest.spyOn(GetArticlesPreviewsApiModule, 'getArticlesPreviews');
  });

  it('should re-fetch and set undefined as articlesPreviews if searchConditions are changed', () => {
    wrapper.setProps({
      searchConditions: {
        category: 1,
      },
    });
    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();
    instance['componentDidUpdate']({
      searchConditions: {},
    } as Props);

    expect(instance.state.articlesPreviews).toBe(undefined);
  });
});
