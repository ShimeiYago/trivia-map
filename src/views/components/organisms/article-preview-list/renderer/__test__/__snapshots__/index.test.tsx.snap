// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests articlesPreviews is zero 1`] = `
<Fragment>
  <div />
  <ForwardRef(Stack)
    spacing={3}
  >
    <ForwardRef(Stack)
      spacing={1}
    >
      <div
        className="pagination-wrapper"
      >
        <ForwardRef(Pagination)
          boundaryCount={1}
          count={2}
          onChange={[Function]}
          page={1}
          siblingCount={0}
          size="large"
        />
      </div>
    </ForwardRef(Stack)>
    <ForwardRef(Typography)
      align="center"
    >
      表示する投稿がありません。
    </ForwardRef(Typography)>
    <ForwardRef(Stack)
      spacing={1}
    >
      <div
        className="pagination-wrapper"
      >
        <ForwardRef(Pagination)
          boundaryCount={1}
          count={2}
          onChange={[Function]}
          page={1}
          siblingCount={0}
          size="large"
        />
      </div>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Fragment>
`;

exports[`Shallow Snapshot Tests basic 1`] = `
<Fragment>
  <div />
  <ForwardRef(Stack)
    spacing={3}
  >
    <CenterSpinner />
  </ForwardRef(Stack)>
</Fragment>
`;

exports[`Shallow Snapshot Tests only one page with large 1`] = `
<Fragment>
  <div />
  <ForwardRef(Stack)
    spacing={3}
  >
    <ForwardRef(Stack)
      spacing={1}
    >
      <ForwardRef(Typography)
        align="center"
        component="div"
        fontSize={16}
      >
        1
        〜
        2
        件 / 全
        2
        件
      </ForwardRef(Typography)>
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      spacing={3}
    >
      <ForwardRef(Box)
        key="preview-list-1"
      >
        <Link
          className="preview-link"
          key="preview-1"
          to="/articles/1"
        >
          <ForwardRef(Card)
            sx={
              Object {
                "&:hover": Object {
                  "transform": "translate(0,-5px)",
                },
                "backgroundColor": "#f5f8fa",
                "transition": "all 0.3s",
              }
            }
          >
            <ForwardRef(CardMedia)
              className="card-media"
              component="img"
              image="https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg"
            />
            <ForwardRef(CardContent)>
              <ForwardRef(Typography)
                component="h3"
                gutterBottom={true}
                sx={
                  Object {
                    "wordBreak": "break-all",
                  }
                }
                variant="h5"
              >
                ノーチラス号のエンジン
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 1,
                  }
                }
              >
                <IconAndText
                  align="left"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="隠れミッキー"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 1,
                  }
                }
              >
                <IconAndText
                  align="left"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="1"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 2,
                  }
                }
              >
                <IconAndText
                  align="right"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="2022/06/16 21:42"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                align="center"
                component="div"
              >
                <IconAndText
                  component="span"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="くわしく読む"
                />
              </ForwardRef(Typography)>
            </ForwardRef(CardContent)>
          </ForwardRef(Card)>
        </Link>
      </ForwardRef(Box)>
      <ForwardRef(Box)
        key="preview-list-2"
      >
        <Link
          className="preview-link"
          key="preview-2"
          to="/articles/2"
        >
          <ForwardRef(Card)
            sx={
              Object {
                "&:hover": Object {
                  "transform": "translate(0,-5px)",
                },
                "backgroundColor": "#f5f8fa",
                "transition": "all 0.3s",
              }
            }
          >
            <ForwardRef(CardMedia)
              className="card-media"
              component="img"
              image="no-image-16x7.jpg"
            />
            <ForwardRef(CardContent)>
              <ForwardRef(Typography)
                component="h3"
                gutterBottom={true}
                sx={
                  Object {
                    "wordBreak": "break-all",
                  }
                }
                variant="h5"
              >
                ノーチラス号の名前の由来
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 1,
                  }
                }
              >
                <IconAndText
                  align="left"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="隠れミッキー"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 1,
                  }
                }
              >
                <IconAndText
                  align="left"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="1"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 2,
                  }
                }
              >
                <IconAndText
                  align="right"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="2022/06/16 21:42"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                align="center"
                component="div"
              >
                <IconAndText
                  component="span"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="くわしく読む"
                />
              </ForwardRef(Typography)>
            </ForwardRef(CardContent)>
          </ForwardRef(Card)>
        </Link>
      </ForwardRef(Box)>
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      spacing={1}
    >
      <ForwardRef(Typography)
        align="center"
        component="div"
        fontSize={16}
      >
        1
        〜
        2
        件 / 全
        2
        件
      </ForwardRef(Typography)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Fragment>
`;

exports[`Shallow Snapshot Tests only one page with popup 1`] = `
<Fragment>
  <div />
  <ForwardRef(Stack)
    spacing={1}
  >
    <ForwardRef(Box)
      maxHeight="240px"
      sx={
        Object {
          "position": "relative",
        }
      }
    >
      <ForwardRef(Box)
        maxHeight="240px"
        sx={
          Object {
            "overflow": "scroll",
          }
        }
      >
        <ForwardRef(Stack)
          spacing={2}
          sx={
            Object {
              "pb": "50px",
            }
          }
        >
          <ForwardRef(Box)
            key="preview-list-1"
          >
            <Link
              className="preview-link"
              key="preview-1"
              to="/articles/1"
            >
              <ForwardRef(Card)
                sx={
                  Object {
                    "&:hover": Object {
                      "transform": "translate(0,-5px)",
                    },
                    "backgroundColor": "#f5f8fa",
                    "p": 1,
                    "transition": "all 0.3s",
                  }
                }
              >
                <ForwardRef(Stack)
                  spacing={1}
                >
                  <DynamicAlignedText
                    component="h2"
                    sx={
                      Object {
                        "wordBreak": "break-all",
                      }
                    }
                    variant="h6"
                  >
                    ノーチラス号のエンジン
                  </DynamicAlignedText>
                  <ForwardRef(Box)
                    textAlign="center"
                  >
                    <Image
                      borderRadius={true}
                      height="100px"
                      objectFit="cover"
                      src="https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg"
                      width="200px"
                    />
                  </ForwardRef(Box)>
                  <ForwardRef(Divider) />
                  <ForwardRef(Box)>
                    <ForwardRef(Typography)
                      component="div"
                      sx={
                        Object {
                          "pr": 2,
                        }
                      }
                    >
                      <IconAndText
                        align="right"
                        fontSize={12}
                        iconComponent={<Memo />}
                        iconPosition="left"
                        text="隠れミッキー"
                      />
                    </ForwardRef(Typography)>
                    <ForwardRef(Typography)
                      component="div"
                      sx={
                        Object {
                          "pr": 2,
                        }
                      }
                    >
                      <IconAndText
                        align="right"
                        fontSize={14}
                        iconComponent={
                          <Memo
                            fontSize="inherit"
                          />
                        }
                        iconPosition="left"
                        text="1"
                      />
                    </ForwardRef(Typography)>
                  </ForwardRef(Box)>
                  <ForwardRef(Typography)
                    align="center"
                  >
                    <IconAndText
                      component="span"
                      iconComponent={<Memo />}
                      iconPosition="left"
                      text="くわしく読む"
                      variant="button"
                    />
                  </ForwardRef(Typography)>
                </ForwardRef(Stack)>
              </ForwardRef(Card)>
            </Link>
          </ForwardRef(Box)>
          <ForwardRef(Box)
            key="preview-list-2"
          >
            <Link
              className="preview-link"
              key="preview-2"
              to="/articles/2"
            >
              <ForwardRef(Card)
                sx={
                  Object {
                    "&:hover": Object {
                      "transform": "translate(0,-5px)",
                    },
                    "backgroundColor": "#f5f8fa",
                    "p": 1,
                    "transition": "all 0.3s",
                  }
                }
              >
                <ForwardRef(Stack)
                  spacing={1}
                >
                  <DynamicAlignedText
                    component="h2"
                    sx={
                      Object {
                        "wordBreak": "break-all",
                      }
                    }
                    variant="h6"
                  >
                    ノーチラス号の名前の由来
                  </DynamicAlignedText>
                  <ForwardRef(Divider) />
                  <ForwardRef(Box)>
                    <ForwardRef(Typography)
                      component="div"
                      sx={
                        Object {
                          "pr": 2,
                        }
                      }
                    >
                      <IconAndText
                        align="right"
                        fontSize={12}
                        iconComponent={<Memo />}
                        iconPosition="left"
                        text="隠れミッキー"
                      />
                    </ForwardRef(Typography)>
                    <ForwardRef(Typography)
                      component="div"
                      sx={
                        Object {
                          "pr": 2,
                        }
                      }
                    >
                      <IconAndText
                        align="right"
                        fontSize={14}
                        iconComponent={
                          <Memo
                            fontSize="inherit"
                          />
                        }
                        iconPosition="left"
                        text="1"
                      />
                    </ForwardRef(Typography)>
                  </ForwardRef(Box)>
                  <ForwardRef(Typography)
                    align="center"
                  >
                    <IconAndText
                      component="span"
                      iconComponent={<Memo />}
                      iconPosition="left"
                      text="くわしく読む"
                      variant="button"
                    />
                  </ForwardRef(Typography)>
                </ForwardRef(Stack)>
              </ForwardRef(Card)>
            </Link>
          </ForwardRef(Box)>
        </ForwardRef(Stack)>
      </ForwardRef(Box)>
      <ForwardRef(Box)
        sx={
          Object {
            "background": "linear-gradient(to bottom, rgba(0,0,0,0), rgba(0,0,0,0.8))",
            "bottom": 0,
            "height": "50px",
            "left": 0,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
    </ForwardRef(Box)>
    <ForwardRef(Stack)
      spacing={1}
    >
      <ForwardRef(Typography)
        align="right"
        component="div"
        fontSize={14}
      >
        1
        〜
        2
        件 / 全
        2
        件
      </ForwardRef(Typography)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Fragment>
`;

exports[`Shallow Snapshot Tests variant popup 1`] = `
<Fragment>
  <div />
  <ForwardRef(Stack)
    spacing={1}
  >
    <ForwardRef(Box)
      maxHeight="240px"
      sx={
        Object {
          "position": "relative",
        }
      }
    >
      <ForwardRef(Box)
        maxHeight="240px"
        sx={
          Object {
            "overflow": "scroll",
          }
        }
      >
        <ForwardRef(Stack)
          spacing={2}
          sx={
            Object {
              "pb": "50px",
            }
          }
        >
          <ForwardRef(Box)
            key="preview-list-1"
          >
            <Link
              className="preview-link"
              key="preview-1"
              to="/articles/1"
            >
              <ForwardRef(Card)
                sx={
                  Object {
                    "&:hover": Object {
                      "transform": "translate(0,-5px)",
                    },
                    "backgroundColor": "#f5f8fa",
                    "p": 1,
                    "transition": "all 0.3s",
                  }
                }
              >
                <ForwardRef(Stack)
                  spacing={1}
                >
                  <DynamicAlignedText
                    component="h2"
                    sx={
                      Object {
                        "wordBreak": "break-all",
                      }
                    }
                    variant="h6"
                  >
                    ノーチラス号のエンジン
                  </DynamicAlignedText>
                  <ForwardRef(Box)
                    textAlign="center"
                  >
                    <Image
                      borderRadius={true}
                      height="100px"
                      objectFit="cover"
                      src="https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg"
                      width="200px"
                    />
                  </ForwardRef(Box)>
                  <ForwardRef(Divider) />
                  <ForwardRef(Box)>
                    <ForwardRef(Typography)
                      component="div"
                      sx={
                        Object {
                          "pr": 2,
                        }
                      }
                    >
                      <IconAndText
                        align="right"
                        fontSize={12}
                        iconComponent={<Memo />}
                        iconPosition="left"
                        text="隠れミッキー"
                      />
                    </ForwardRef(Typography)>
                    <ForwardRef(Typography)
                      component="div"
                      sx={
                        Object {
                          "pr": 2,
                        }
                      }
                    >
                      <IconAndText
                        align="right"
                        fontSize={14}
                        iconComponent={
                          <Memo
                            fontSize="inherit"
                          />
                        }
                        iconPosition="left"
                        text="1"
                      />
                    </ForwardRef(Typography)>
                  </ForwardRef(Box)>
                  <ForwardRef(Typography)
                    align="center"
                  >
                    <IconAndText
                      component="span"
                      iconComponent={<Memo />}
                      iconPosition="left"
                      text="くわしく読む"
                      variant="button"
                    />
                  </ForwardRef(Typography)>
                </ForwardRef(Stack)>
              </ForwardRef(Card)>
            </Link>
          </ForwardRef(Box)>
          <ForwardRef(Box)
            key="preview-list-2"
          >
            <Link
              className="preview-link"
              key="preview-2"
              to="/articles/2"
            >
              <ForwardRef(Card)
                sx={
                  Object {
                    "&:hover": Object {
                      "transform": "translate(0,-5px)",
                    },
                    "backgroundColor": "#f5f8fa",
                    "p": 1,
                    "transition": "all 0.3s",
                  }
                }
              >
                <ForwardRef(Stack)
                  spacing={1}
                >
                  <DynamicAlignedText
                    component="h2"
                    sx={
                      Object {
                        "wordBreak": "break-all",
                      }
                    }
                    variant="h6"
                  >
                    ノーチラス号の名前の由来
                  </DynamicAlignedText>
                  <ForwardRef(Divider) />
                  <ForwardRef(Box)>
                    <ForwardRef(Typography)
                      component="div"
                      sx={
                        Object {
                          "pr": 2,
                        }
                      }
                    >
                      <IconAndText
                        align="right"
                        fontSize={12}
                        iconComponent={<Memo />}
                        iconPosition="left"
                        text="隠れミッキー"
                      />
                    </ForwardRef(Typography)>
                    <ForwardRef(Typography)
                      component="div"
                      sx={
                        Object {
                          "pr": 2,
                        }
                      }
                    >
                      <IconAndText
                        align="right"
                        fontSize={14}
                        iconComponent={
                          <Memo
                            fontSize="inherit"
                          />
                        }
                        iconPosition="left"
                        text="1"
                      />
                    </ForwardRef(Typography)>
                  </ForwardRef(Box)>
                  <ForwardRef(Typography)
                    align="center"
                  >
                    <IconAndText
                      component="span"
                      iconComponent={<Memo />}
                      iconPosition="left"
                      text="くわしく読む"
                      variant="button"
                    />
                  </ForwardRef(Typography)>
                </ForwardRef(Stack)>
              </ForwardRef(Card)>
            </Link>
          </ForwardRef(Box)>
        </ForwardRef(Stack)>
      </ForwardRef(Box)>
      <ForwardRef(Box)
        sx={
          Object {
            "background": "linear-gradient(to bottom, rgba(0,0,0,0), rgba(0,0,0,0.8))",
            "bottom": 0,
            "height": "50px",
            "left": 0,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
    </ForwardRef(Box)>
    <ForwardRef(Stack)
      spacing={1}
    >
      <div
        className="pagination-wrapper"
      >
        <ForwardRef(Pagination)
          boundaryCount={1}
          count={2}
          onChange={[Function]}
          page={1}
          siblingCount={0}
          size="small"
        />
      </div>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Fragment>
`;

exports[`Shallow Snapshot Tests variant sidebar 1`] = `
<Fragment>
  <div />
  <ForwardRef(Stack)
    spacing={3}
  >
    <ForwardRef(Stack)
      spacing={1}
    >
      <ForwardRef(Box)
        key="preview-list-1"
      >
        <Link
          className="preview-link"
          key="preview-1"
          to="/articles/1"
        >
          <ForwardRef(Card)
            sx={
              Object {
                "&:hover": Object {
                  "transform": "translate(0,-5px)",
                },
                "backgroundColor": "#f5f8fa",
                "transition": "all 0.3s",
              }
            }
          >
            <ForwardRef(CardContent)>
              <ForwardRef(Grid)
                container={true}
                spacing={1}
              >
                <ForwardRef(Grid)
                  item={true}
                  xs={5}
                >
                  <Image
                    height="100px"
                    objectFit="cover"
                    src="https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg"
                    width="full"
                  />
                </ForwardRef(Grid)>
                <ForwardRef(Grid)
                  item={true}
                  xs={7}
                >
                  <ForwardRef(Typography)
                    component="h3"
                    gutterBottom={true}
                    sx={
                      Object {
                        "wordBreak": "break-all",
                      }
                    }
                    variant="body1"
                  >
                    ノーチラス号のエンジン
                  </ForwardRef(Typography)>
                </ForwardRef(Grid)>
              </ForwardRef(Grid)>
            </ForwardRef(CardContent)>
          </ForwardRef(Card)>
        </Link>
      </ForwardRef(Box)>
      <ForwardRef(Box)
        key="preview-list-2"
      >
        <Link
          className="preview-link"
          key="preview-2"
          to="/articles/2"
        >
          <ForwardRef(Card)
            sx={
              Object {
                "&:hover": Object {
                  "transform": "translate(0,-5px)",
                },
                "backgroundColor": "#f5f8fa",
                "transition": "all 0.3s",
              }
            }
          >
            <ForwardRef(CardContent)>
              <ForwardRef(Grid)
                container={true}
                spacing={1}
              >
                <ForwardRef(Grid)
                  item={true}
                  xs={5}
                >
                  <Image
                    height="100px"
                    objectFit="cover"
                    src="no-image-16x7.jpg"
                    width="full"
                  />
                </ForwardRef(Grid)>
                <ForwardRef(Grid)
                  item={true}
                  xs={7}
                >
                  <ForwardRef(Typography)
                    component="h3"
                    gutterBottom={true}
                    sx={
                      Object {
                        "wordBreak": "break-all",
                      }
                    }
                    variant="body1"
                  >
                    ノーチラス号の名前の由来
                  </ForwardRef(Typography)>
                </ForwardRef(Grid)>
              </ForwardRef(Grid)>
            </ForwardRef(CardContent)>
          </ForwardRef(Card)>
        </Link>
      </ForwardRef(Box)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Fragment>
`;

exports[`Shallow Snapshot Tests with articlesPreviews 1`] = `
<Fragment>
  <div />
  <ForwardRef(Stack)
    spacing={3}
  >
    <ForwardRef(Stack)
      spacing={1}
    >
      <div
        className="pagination-wrapper"
      >
        <ForwardRef(Pagination)
          boundaryCount={1}
          count={2}
          onChange={[Function]}
          page={1}
          siblingCount={0}
          size="large"
        />
      </div>
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      spacing={3}
    >
      <ForwardRef(Box)
        key="preview-list-1"
      >
        <Link
          className="preview-link"
          key="preview-1"
          to="/articles/1"
        >
          <ForwardRef(Card)
            sx={
              Object {
                "&:hover": Object {
                  "transform": "translate(0,-5px)",
                },
                "backgroundColor": "#f5f8fa",
                "transition": "all 0.3s",
              }
            }
          >
            <ForwardRef(CardMedia)
              className="card-media"
              component="img"
              image="https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg"
            />
            <ForwardRef(CardContent)>
              <ForwardRef(Typography)
                component="h3"
                gutterBottom={true}
                sx={
                  Object {
                    "wordBreak": "break-all",
                  }
                }
                variant="h5"
              >
                ノーチラス号のエンジン
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 1,
                  }
                }
              >
                <IconAndText
                  align="left"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="隠れミッキー"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 1,
                  }
                }
              >
                <IconAndText
                  align="left"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="1"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 2,
                  }
                }
              >
                <IconAndText
                  align="right"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="2022/06/16 21:42"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                align="center"
                component="div"
              >
                <IconAndText
                  component="span"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="くわしく読む"
                />
              </ForwardRef(Typography)>
            </ForwardRef(CardContent)>
          </ForwardRef(Card)>
        </Link>
      </ForwardRef(Box)>
      <ForwardRef(Box)
        key="preview-list-2"
      >
        <Link
          className="preview-link"
          key="preview-2"
          to="/articles/2"
        >
          <ForwardRef(Card)
            sx={
              Object {
                "&:hover": Object {
                  "transform": "translate(0,-5px)",
                },
                "backgroundColor": "#f5f8fa",
                "transition": "all 0.3s",
              }
            }
          >
            <ForwardRef(CardMedia)
              className="card-media"
              component="img"
              image="no-image-16x7.jpg"
            />
            <ForwardRef(CardContent)>
              <ForwardRef(Typography)
                component="h3"
                gutterBottom={true}
                sx={
                  Object {
                    "wordBreak": "break-all",
                  }
                }
                variant="h5"
              >
                ノーチラス号の名前の由来
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 1,
                  }
                }
              >
                <IconAndText
                  align="left"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="隠れミッキー"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 1,
                  }
                }
              >
                <IconAndText
                  align="left"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="1"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "mb": 2,
                  }
                }
              >
                <IconAndText
                  align="right"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="2022/06/16 21:42"
                />
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                align="center"
                component="div"
              >
                <IconAndText
                  component="span"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="くわしく読む"
                />
              </ForwardRef(Typography)>
            </ForwardRef(CardContent)>
          </ForwardRef(Card)>
        </Link>
      </ForwardRef(Box)>
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      spacing={1}
    >
      <div
        className="pagination-wrapper"
      >
        <ForwardRef(Pagination)
          boundaryCount={1}
          count={2}
          onChange={[Function]}
          page={1}
          siblingCount={0}
          size="large"
        />
      </div>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Fragment>
`;
